src/poll.rs

```diff
@@ -503,12 +506,25 @@ mod syscall {
     }
 
     /// Safe wrapper around the `poll` system call.
-    pub(super) fn poll(fds: &mut [PollFd<'_>], timeout: i32) -> io::Result<usize> {
+    pub(super) fn poll(fds: &mut [PollFd<'_>], timeout: Option<Duration>) -> io::Result<usize> {
+        let timeout_ms = match timeout {
+            Some(timeout) => {
+                // Round up to a whole millisecond.
+                let mut ms = timeout.as_millis().try_into().unwrap_or(std::u64::MAX);
+                if Duration::from_millis(ms) < timeout {
+                    ms = ms.saturating_add(1);
+                }
+                ms.try_into()
+                    .map_err(|_| io::Error::from(io::ErrorKind::InvalidInput))?
```

Comment:
Same here.

Reply:
@sunfishcode This does not seem to have been addressed yet. Or am I missing something?

Reply:
Ah, you're right, I had missed that one. It's now addressed.